lib/adapter.js

```diff
@@ -0,0 +1,494 @@
+/*******************************************************************************
+ * Copyright 2014, 2015, 2016 IBM Corp.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ ******************************************************************************/
+var appmetrics = require('appmetrics');
+var os = require('os');
+var path = require('path');
+
+appmetrics.enable('eventloop');
+var amapi = appmetrics.monitor();
+
+// Global objects to store metric data
+var eventLoopMetrics = {count: 0, averageRunningTotal: 0};
+var cpuMetrics = {system: 0, user: 0, count: 0};
+var httpMetrics = {count: 0};
+var probeList = ["leveldown", "memcached", "mongo", "mysql", "oracle", "oracledb", "postgres", "redis", "socketio", "strong-oracle"];
+var probeMetrics = {};
+var heapMetrics = {};
+var gcMetrics = {used: 0, count: 0};
+var messageMetrics = {};
+
+// Keep track of Cpu Profiling
+var profiling = false;
+var objectTrackingEnabled = false;
+var lastObjectHistogram = {};
+
+// Callback interval
+var callbackInterval = 10000;
+var objectTrackingCallbackInterval = 15000;
+
+// ---------------------------------------------
+// Public functions to be exported
+// ---------------------------------------------
+
+// Metrics API Use function
+function use(callback) {
+    // Set up listeners for all the metrics here
+    initialise();
+
+    // The callback needs to fire every 60 seconds with data from the last 60
+	// seconds
+    setInterval(function() {
+        // Get data from last 60 seconds
+        var eventLoopData = getEventLoopData();
+        var cpuData = getCpuData();
+        var httpData = getHTTPData();
+		var probeData;
```

Comment:
please define your vars closer to where they are used, it makes it easier to read the code without paging back and forth